src/interpreter/mod.rs

```diff
@@ -37,17 +37,6 @@ struct GlobalEvalContext<'a, 'tcx: 'a> {
     /// The virtual memory system.
     memory: Memory,
 
-    /// Another stack containing the type substitutions for the current function invocation. It
-    /// exists separately from `stack` because it must contain the `Substs` for a function while
-    /// *creating* the `Frame` for that same function.
-    substs_stack: Vec<&'tcx Substs<'tcx>>,
-
-    // TODO(solson): Merge with `substs_stack`. Also try restructuring `Frame` to accomodate.
-    /// A stack of the things necessary to print good strack traces:
-    ///   * Function DefIds and Substs to print proper substituted function names.
-    ///   * Spans pointing to specific function calls in the source.
-    name_stack: Vec<(DefId, &'tcx Substs<'tcx>, codemap::Span)>,
-
```

Comment:
Hmm, I guess cleaning up the stacks was a lot easier than I thought (particularly `substs_stack`).

We definitely need a ton more tests dealing with generic functions, monomorphization, and substs to make sure we haven't screwed it up, but your change here seems correct to me.

Thanks for the cleanups. \o/